doc/topics/writing-recipes.rst

```diff
@@ -0,0 +1,355 @@
+  second phase being the responsibility of either the ``install`` or
+  ``update`` methods.
+
+- The ``install`` method installs new parts.
+
+- The ``update`` method updates previously installed parts.  It's
+  often an empty method or an alias for ``install``.
+
+Buildout phases
+---------------
+
+When buildout is run using the default :ref:`install command
+<install-command>`, parts are installed in several phases:
+
+1. Parts are initialized by calling their recipe constructors.  This may
+   cause part configuration options to be updated, as described below.
+
+2. Part options are compared to part options from previous runs
+   [#installed]_.
+
+   - Parts from previous runs that are no longer part of the buildout
+     are uninstalled.
+
+   - Parts from previous runs whose options have changed are also
+     uninstalled.
+
+3. Parts are either installed or updated.
+
+   ``install()`` is called on new parts or old parts that were uninstalled.
+
+   ``update()`` is called on old parts whose configuration hasn't changed.
+
+
+Initialization phase: the constructor
+-------------------------------------
+
+The constructor is passed 3 arguments:
+
+``buildout``
+   The buildout configuration
+
+   The buildout configuration is a mapping from section names to
+   sections.  Sections, are mappings from option names to values.  The
+   buildout configuration allows the recipe to access configuration
+   data in much the same way as configuration files use :ref:`value
+   substitutions <value-substitutions>`.
+
+``name``
+   The name of the section the recipe was used for
+
+``options``
+   The part options
+
+   This is a mapping object and may be written to to save derived
+   configuration, to provide information for use by other part
+   recipes, or for :ref:`value substitutions <value-substitutions>`.
+
+Nothing should be installed in this phase.
+
+If the part being installed isn't new, options after calling the
+constructor are compared to the options from the previous Buildout
+run. If they are different, then the part will be uninstalled and then
+re-installed by calling the ``install`` method, otherwise, the ``update``
+method will be called.
+
+Install or update phase
+-----------------------
+
+In this phase, ``install()`` or ``update()`` is called, depending on
+whether the part is new or has new configuration.
+
+This is the phase in which the part does its work.  In addition to
+affecting changes, these methods have some responsibilities that can
+be a little delicate:
+
+- If an error is raised, it is the responsibility of the recipe to
+  undo any partial changes.
+
+- If the recipe created any files or directories, the recipe should
+  return their paths.  Doing so allows Buildout to take care of
+  removing them if the part is uninstalled, making a separate
+  uninstall recipe unnecessary.
+
+To make these responsibilities easier to cope with, the ``option``
```

Comment:
Is this the same as the `options` object, which was mentioned as the 3rd argument passed to the constructor?